tests/integration/multi_table/test_hma.py

```diff
@@ -31,3 +35,44 @@ def test_hma(tmpdir):
 
     for normal_table, increased_table in zip(normal_sample.values(), increased_sample.values()):
         assert increased_table.size > normal_table.size
+
+
+def test_hma_reset_sampling(tmpdir):
+    """End to end integration test that uses ``reset_sampling``.
+
+    This test uses ``reset_sampling`` to ensure that the model will generate the same data
+    as the first sample after this method has been called.
+    """
+    # Setup
+    faker = Faker()
+    metadata, data = load_demo(metadata=True)
+    metadata.to_json(tmpdir / 'old.json')
+    MultiTableMetadata.upgrade_metadata(tmpdir / 'old.json', tmpdir / 'new.json')
```

Comment:
instead if doing this can we use the new [download_demo](https://github.com/sdv-dev/SDV/blob/V1.0.0.dev/sdv/datasets/demo.py#L98) method and avoid conversions?

Reply:
I think we will have to doublecheck the datasets before doing that.

Reply:
The datasets should be good. The problem is we're going to remove the old code so we can't use the old `load_demo`